docs/source/ext/database.rst

```diff
@@ -335,9 +335,9 @@ Furthermore, it is in charge of assigning the ``persistent_id`` when an entity i
 
 WorkflowBuilder
 ^^^^^^^^^^^^^^^
-The ``WorkflowBuilder`` class loads the steps and ports of an existing workflow from a ``Database`` and inserts them into a new workflow object received as a constructor argument. It extends the ``DefaultDatabaseLoadingContext`` class and overrides only the methods involving ``step``, ``port``, and ``workflow`` entities. In particular, the ``add_*`` methods of these entities must not set the ``persistent_id``, as they are dealing with a newly-created workflow, and the ``load_*`` methods should reset the internal state of their entities to the initial value (e.g., reset the status to `Status.WAITING` and clear the `terminated` flag).
-
-The ``load_workflow`` method must behave in two different ways, depending on whether it is called directly from a user or in the internal logic of another entity's ``load`` method. In the first case, it should load all the entities related to the original workflow, identified by the ``persistent_id`` argument, into the new one. In the latter case it should simply return the new workflow entity being built.
+The ``WorkflowBuilder`` class loads the steps and ports of an existing workflow from a ``Database`` and inserts them into a new workflow object. The class extends the ``DefaultDatabaseLoadingContext`` class and overrides only the methods involving ``step``, ``port``, and ``workflow`` entities. In particular, the ``add_*`` methods of these entities must not set the ``persistent_id``, as they are dealing with a newly-created workflow, and the ``load_*`` methods should reset the internal state of their entities to the initial value (e.g., reset the status to `Status.WAITING` and clear the `terminated` flag).
+If the deep copy attribute is set in the ``WorkflowBuilder``, the builder will return a new workflow duplicating all steps and ports instances. Otherwise, a new instance of the workflow is returned, but without loading the steps and ports.
```

Comment:
```suggestion
If a ``WorkflowBuilder`` object is constructed with  the ``deep_copy`` attribute equal to ``True``, the builder will return a new workflow duplicating all steps and ports instances. Otherwise, a new instance of the workflow is returned, but without loading the steps and ports.
```